api/src/services/translation.service.ts

```diff
@@ -192,9 +194,39 @@ export class TranslationService {
     listing: Listing,
     language: LanguagesEnum,
   ) {
-    return this.prisma.generatedListingTranslations.findFirst({
-      where: { listingId: listing.id, language: language },
+    const res = await this.prisma.generatedListingTranslations.findFirst({
+      where: {
+        listingId: listing.id,
+        language: language,
+      },
+    });
+
+    //check if associated preferences have changed since translation creation
+    let mostRecentPreferenceUpdate = new Date(2000, 1);
+    listing.listingMultiselectQuestions?.forEach((multiselectObj) => {
+      const multiselectUpdatedAt =
+        multiselectObj.multiselectQuestions?.updatedAt;
+      if (mostRecentPreferenceUpdate < multiselectUpdatedAt) {
+        mostRecentPreferenceUpdate = multiselectUpdatedAt;
+      }
     });
+    // refresh translations if preference recently updated or translations are a day old
+    const yesterday = new Date();
+    yesterday.setDate(yesterday.getDate() - 1);
```

Comment:
Totally understand I think there was some offline convo about this, but it seems like this would result in us re-translating more often than we would need to?

Reply:
Yeahhh I share that worry. It seems like the most sustainable solution is to shift the way we handle translations to be triggered within the listing service (ie. on update) as opposed to when the user selects a language for the application. Feel free to add additional context @YazeedLoonat but he was also making the point that google translations are updated frequently and this temporary solution would help to provide the best experience for the user and that the added cost is worth that benefit. At this point, I think it'd be best to loop in product to get their thoughts on this question.

Reply:
I also think we might want to discuss this more. I don't have all of the context, but getting new translations every day seems like it could be more excessive than need be